File: 217.存在重复元素.cpp
/*
 * @lc app=leetcode.cn id=217 lang=cpp
 *
 * [217] 存在重复元素
 */

// @lc code=start
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> m;
        for(auto i : nums) {
            if(m.count(i)) {
                return true;
            }
            m.insert(i);
        }
        return false;
    }
};
// @lc code=end

File: 217.存在重复元素.go
/*
 * @lc app=leetcode.cn id=217 lang=golang
 *
 * [217] 存在重复元素
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
	m := map[int]bool {}
	for _, n := range nums {
		if m[n] {
			return true
	} else {
		m[n] = true
	}
	}
	return false
}
// @lc code=end

